magenta/music/musicxml_parser.py

```diff
@@ -0,0 +1,678 @@
+    score_str = "ScorePart: " + self.part_name
+    score_str += ", Channel: " + str(self.midi_channel)
+    score_str += ", Program: " + str(self.midi_program)
+    return score_str
+
+
+class Part(object):
+  """Internal represention of a MusicXML <part> element"""
+  def __init__(self, xml_part, score_part, state):
+    self.xml_part = xml_part
+    self.score_part = score_part
+    self.measures = []
+    self.transposes = False
+    self.state = state
+    self.parse()
+
+  def parse(self):
+    """Parse the <part> element"""
+
+    # Reset the time position when parsing each part
+    #CURRENT_TIME_POSITION = 0
+    self.state.time_position = 0
+    self.state.midi_channel = self.score_part.midi_channel
+    self.state.midi_program = self.score_part.midi_program
+    self.state.transpose = 0
+
+    xml_measures = self.xml_part.findall("measure")
+    for child in xml_measures:
+      measure = Measure(child, self, self.state)
+      self.measures.append(measure)
+
+  def getnotesinvoice(self, voice):
+    """Get all the notes in this part for the specified voice number"""
+    notes = []
+    for measure in self.measures:
+      for note in measure.notes:
+        if note.voice == voice:
+          notes.append(note)
+
+    return notes
+
+  def __str__(self):
+    part_str = "Part: " + self.score_part.part_name
+    return part_str
+
+
+class Measure(object):
+  """Internal represention of the MusicXML <measure> element"""
+  def __init__(self, xml_measure, part, state):
+    self.xml_measure = xml_measure
+    self.notes = []
+    self.tempos = []
+    self.time_signature = None
+    self.key_signature = KeySignature(state) # Default to C major
+    self.current_ticks = 0      # Cumulative tick counter for this measure
+    self.transpose = 0          # Number of semitones to transpose notes
+    self.part = part
+    self.state = state
+    self.parse()
+
+  def parse(self):
+    """Parse the <measure> element"""
+
+    for child in self.xml_measure:
+      if child.tag == "attributes":
+        self.parseattributes(child)
+      elif child.tag == "backup":
+        self.parsebackup(child)
+      elif child.tag == "direction":
+        self.parsedirection(child)
+      elif child.tag == "forward":
+        self.parseforward(child)
+      elif child.tag == "note":
+        note = Note(child, self.state)
+        self.notes.append(note)
+        # Keep track of current note as previous note for chord timings
+        self.state.previous_note = note
+
+  def parseattributes(self, xml_attributes):
+    """Parse the MusicXML <attributes> element"""
+
+    for child in xml_attributes:
+      if child.tag == "divisions":
+        #CURRENT_DIVISIONS = int(child.text)
+        self.state.divisions = int(child.text)
+      elif child.tag == "key":
+        self.key_signature = KeySignature(self.state, child)
+      elif child.tag == "time":
+        self.time_signature = TimeSignature(self.state, child)
+      elif child.tag == "transpose":
```

Comment:
Ah, I see. Could you add a comment explaining that, just so it's clear that there are other valid tags and we're just choosing to ignore them. Maybe something like:

```
else:
  # Ignore other tag types because they are not relevant to Magenta.
  pass
```